src/main/java/org/jitsi/jigasi/transcription/AbstractTranscriptPublisher.java

```diff
@@ -194,15 +195,45 @@ protected void sendMessage(ChatRoom chatRoom, T message)
         try
         {
             chatRoom.sendMessage(chatRoomMessage);
-            if (logger.isDebugEnabled())
-                logger.debug("Sending message: \"" + messageString + "\"");
+            if (logger.isTraceEnabled())
+                logger.trace("Sending message: \"" + messageString + "\"");
         }
         catch (OperationFailedException e)
         {
             logger.warn("Failed to send message " + messageString, e);
         }
     }
 
+    /**
+     * Send a json-message to the muc room
+     *
+     * @param chatRoom the chatroom to send the message to
+     * @param jsonMessage the json message to send
+     */
+    protected void sendJsonMessage(ChatRoom chatRoom, T jsonMessage)
+    {
+        if (chatRoom == null)
+        {
+            logger.error("Cannot sent message as chatRoom is null");
+            return;
+        }
+
+        String messageString = jsonMessage.toString();
+        //Creates a Message without a body.
+        Message chatRoomMessage = chatRoom.createMessage(null);
+        try
+        {
+            if(chatRoom instanceof ChatRoomJabberImpl)
+                ((ChatRoomJabberImpl)chatRoom).sendJsonMessage(chatRoomMessage,
+                        messageString);
+            if (logger.isTraceEnabled())
+                logger.trace("Sending json message: \"" + messageString + "\"");
+        }
+        catch (OperationFailedException e)
+        {
+            logger.warn("Failed to send json message " + messageString, e);
```

Comment:
with the current PR in jitsi this will result in 1 error and 1 warning log message for the same failure of not sending?

Reply:
Removed it from Jitsi PR.